tests/test_bashcomplete.py

```diff
@@ -407,3 +410,34 @@ def hsub():
     # If the user exactly types out the hidden command, complete its subcommands.
     assert choices_without_help(cli, ['hgroup'], '') == ['hgroupsub']
     assert choices_without_help(cli, ['hsub'], '--h') == ['--hname']
+
+def test_args_with_double_dash_complete():
+    def _complete(ctx, args, incomplete):
+        _tmp =  ['abc', 'def', 'ghi', 'jkl', 'mno', 'pqr', 'stu', 'vwx', 'yz']
+        _tmp += ['-abc', '--def', '-ghi', '--jkl', '-osk', '--oab']
+        return [x for x in _tmp if x.startswith(incomplete)]
+
+    @click.group()
+    def entrypoint():
+        pass
+
+    @click.command()
+    @click.option('--opt')
+    @click.argument('arg', nargs=-1, autocompletion=_complete)
+    def subcommand(opt, arg):
+        pass
+
+    entrypoint.add_command(subcommand)
+
+    _prod = itertools.product
+
+    mustBeOpt = _prod([['subcommand']],[[''],['abc']],['-','--','--o'])
```

Comment:
no camelCase please. also, pep8-compatible formatting. better wrap a statement on multiple lines than removing whitespace and using strange aliases (like `_prod`) just to keep the lines shortner